client/src/pages/ViewSubmissions.js

```diff
@@ -0,0 +1,39 @@
+import React from 'react';
+import { Image } from './Image';
+
+export class ViewSubmissions extends React.Component {
+	constructor(props) {
+		super(props)
+		this.state = {
+			submissions: [], 
+			submissionKeys: []
+		}
+   	this.viewAllSubmissions = this.viewAllSubmissions.bind(this)
+	}
+
+	componentDidMount() {
+    this.viewAllSubmissions()
+  	}
+
+  	viewAllSubmissions() {
+    	fetch('/contests/2/submissions').then(response =>
+      	response.json().then(data => {
+        	this.setState({ submissions: data.files, submissionKeys: data.fileKeys }); 
+      		})
+    	) 
+  	}
+
+	render() {
+		let imagesArray = []
+		for(let i=0; i < this.state.submissions.length; i++) {
```

Comment:
you probably want to rewrite this to something a bit more "React-y"! by using .map like

```
return (
  {this.state.submissions.map( (submission, i) => (
    <Img imageURL={submission} imageKey={submission} key={i} />
  ))}
)
```

Reply:
I actually changed this part yesterday to exactly what you've suggested haha! So it'll be there with the next commit 👍